engine.js

```diff
@@ -0,0 +1,96 @@
+"format cjs";
+
+var wrap = require('word-wrap');
+var cloneDeep = require('lodash.clonedeep');
+var map = require('lodash.map');
+var longest = require('longest');
+var rightPad = require('right-pad');
+
+// This can be any kind of SystemJS compatible module.
+// We use Commonjs here, but ES6 or AMD would do just
+// fine.
+module.exports = function (options) {
+
+  options = cloneDeep(options || {});
```

Comment:
@LinusU Sure, you seemed (rightly) concerned about modification, so this would protect us against any future changes that could modify.